app/code/core/Mage/Catalog/Model/Resource/Product/Option/Collection.php

```diff
@@ -149,8 +149,9 @@ public function addValuesToResult($storeId = null)
             $optionIds[] = $option->getId();
         }
         if (!empty($optionIds)) {
-            $values = Mage::getModel('catalog/product_option_value')
-                ->getCollection()
+            /** @var Mage_Catalog_Model_Resource_Product_Option_Value_Collection $values */
+            $values = Mage::getModel('catalog/product_option_value')->getCollection();
+            $values
```

Comment:
No need to instantiate model but may be not BC. 
```suggestion   
            $collection = Mage::getResourceModel('catalog/product_option_value_collection');
            $collection
```

Reply:
I don't know, probably nobody did an override on that class and the getcollection method but this type of code is all over the code, I wouldn't change it in this specific PR